api/src/policies/notifications-policy.ts

```diff
@@ -0,0 +1,45 @@
+import { Attributes, FindOptions } from "@sequelize/core"
+
+import { Path } from "@/utils/deep-pick"
+import { User, Notification } from "@/models"
+import { PolicyFactory } from "@/policies/base-policy"
+
+export class NotificationsPolicy extends PolicyFactory(Notification) {
+  show(): boolean {
+    if (this.record.userId === this.user.id) return true
+
+    return false
+  }
+
+  create(): boolean {
+    if (this.user.isSystemAdmin) return true
+
+    return false
+  }
+
+  update(): boolean {
+    if (this.record.userId === this.user.id) return true
+
+    return false
+  }
+
+  destroy(): boolean {
+    if (this.user.isSystemAdmin) return true
+
+    return false
+  }
+
+  permittedAttributes(): Path[] {
+    return ["isRead"]
+  }
```

Comment:
I think you added a comment somewhere about this being a state change and maybe not permitting any updates except by posting to said endpoint? I think that's probably a good idea?
e.g.
POST /notifications/:notificationId/read -> notifications/ReadController#create
DELETE /notifications/:notificationId/read -> notifications/ReadController#destroy (mark unread)

Reply:
I think it makes sense to do this here, since you've migrated the isRead -> readAt. If you leave it half way, the way of setting it in the back end is confusing .

For the `node-vue-template`, my preference is that it should never be containe code that we aren't using in production somewhere, as that better proves out the patterns in practice.